xml_converter/generators/code_generator.py

```diff
@@ -287,49 +293,55 @@ def write_cpp_classes(self, output_directory: str) -> None:
         file_loader = FileSystemLoader('cpp_templates')
         env = Environment(loader=file_loader)
         attribute_names: Dict[str, str] = {}
-        template: Dict[str, Any] = {
-            "header_template": env.get_template("class_template.hpp"),
-            "code_template": env.get_template("class_template.cpp"),
-        }
+        header_template: Template = env.get_template("class_template.hpp")
+        code_template: Template = env.get_template("class_template.cpp")
+        attributes_of_type_marker_category: List[str] = []
         cpp_classes: Dict[str, str] = {
-            "Category": "MarkerCategory", 
-            "Icon": "POI", 
+            "Category": "MarkerCategory",
+            "Icon": "POI",
             "Trail": "Trail"
         }
+
         for filepath in self.data.keys():
             filename = os.path.basename(filepath)
             attribute_names[filepath] = filename.split(".md")[0]
 
         for cpp_class in cpp_classes:
             metadata: Dict[str, SchemaType] = {}
+            attributes_of_type_marker_category = []
 
             for attribute_name in attribute_names:
                 metadata[attribute_name] = self.data[attribute_name].metadata
-                
 
-            attribute_variables, cpp_include_paths = self.generate_cpp_variable_data(metadata, cpp_class, attribute_names)
+            template_variables, cpp_include_paths = self.generate_cpp_variable_data(metadata, cpp_class, attribute_names)
+
+            for template_variable in template_variables:
+                if template_variable[2] == "marker_category":
+                    attributes_of_type_marker_category.append(template_variable[0])
 
             with open(os.path.join(output_directory, lowercase(cpp_class) + ".hpp"), 'w') as f:
-                f.write(template["header_template"].render(
+                f.write(header_template.render(
                     cpp_class=cpp_class,
-                    attribute_variables=sorted(attribute_variables),
+                    template_variables=sorted(template_variables),
                     cpp_include_paths=sorted(cpp_include_paths),
+                    attributes_of_type_marker_category=attributes_of_type_marker_category,
                 ))
 
             with open(os.path.join(output_directory, lowercase(cpp_class) + ".cpp"), 'w') as f:
-                f.write(template["code_template"].render(
+                f.write(code_template.render(
                     cpp_class=cpp_class,
                     cpp_class_header=lowercase(cpp_class),
                     xml_class_name=cpp_classes[cpp_class],
-                    attribute_variables=sorted(attribute_variables),
+                    template_variables=sorted(template_variables),
                     enumerate=enumerate,
+                    attributes_of_type_marker_category=attributes_of_type_marker_category,
                 ))
-
+            print(attributes_of_type_marker_category)
```

Comment:
Did you leave this debugging line in by accident?

Reply:
I did. Removed